sympy/core/add.py

```diff
@@ -996,20 +996,20 @@ def _eval_as_leading_term(self, x, logx=None, cdir=0):
         from sympy.series.order import Order
         from sympy.functions.elementary.exponential import log
         from sympy.functions.elementary.piecewise import Piecewise, piecewise_fold
-        from .function import expand_mul
+        from .function import expand_log, expand_mul
 
-        old = self
+        o = self.getO()
+        if o is None:
+            o = Order(0)
+        old = self.removeO()
 
         if old.has(Piecewise):
             old = piecewise_fold(old)
 
         # This expansion is the last part of expand_log. expand_log also calls
         # expand_mul with factor=True, which would be more expensive
         if any(isinstance(a, log) for a in self.args):
-            logflags = dict(deep=True, log=True, mul=False, power_exp=False,
-                power_base=False, multinomial=False, basic=False, force=False,
-                factor=False)
-            old = old.expand(**logflags)
+            old = expand_log(old)
```

Comment:
Is there any reason for using `expand_log`? The above comment should be removed if such a case is there.

Reply:
No not really , had introduced that change to see if it helps in fixing bugs in `add._eval_as_leading_term()`, had planned to change it back after I get some clarity( done in latest commit)